the-collab-lab/tcl-8-smart-shopping-list
# LB-ALT-create-newList

## Description

For this issue, we created a new `Welcome` component that renders a button to generate new token and save it to the localStorage. The component checks if the user visits the app for the first time by checking the token's existence. If it is the first time, the user is presented with the button to create a new list. Otherwise redirected to the existing list. 

We also created a `token` file that generates tokens for users. 
Tokens are generated using the following approach found [here](https://gist.github.com/segdeha/21a42618ce5a54916c5b58d36ec2992e).

## Related Issue

Closes #3 

## Acceptance Criteria

- [x] For users who do not already have a token/list, a button or link exists at the / route (i.e. home screen) that allows them to create a new list

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

### After

<img width="384" alt="Screen Shot 2020-06-10 at 5 54 57 PM" src="https://user-images.githubusercontent.com/45564632/84322837-afdee380-ab43-11ea-840e-f39e5aac7233.png">

## Testing Steps / QA Criteria

- To test a new functionality, visit the landing page (`/` route) and press the `New List` button to create a new token. 
- Next, open your developer tools, navigate to Applications - > Local Storage, and see your new token.
